add storage resource to volume Relationships

Lauferin opened the pull request:
The Volume detail page shows to which storage it belongs and a lot of other properties.

I added to the Relationships detail the Storage Resource it belongs to (which its respective link).

<img width="691" alt="Screen Shot 2022-07-25 at 14 05 39" src="https://user-images.githubusercontent.com/50288766/180781762-95fbd9b9-2d91-4e0e-9ff0-6db9590a6b4a.png">
<img width="673" alt="Screen Shot 2022-07-25 at 14 06 27" src="https://user-images.githubusercontent.com/50288766/180781771-7f089663-cec9-45cd-aea0-69e2f565d4ce.png">

Fryguy commented:
@jeffibm Can you take a look at this one?

jeffibm commented:
I made the changes in local env and when I opened the volume summary page I see this. probably because of the record is not available for me, but should this be handled?

<img width="1788" alt="image" src="https://user-images.githubusercontent.com/87487049/183035845-d625dc48-cc14-4dca-9569-b6c26b6a5caf.png">


```
[----] F, [2022-08-05T13:53:01.417818 #38950:b6ff8] FATAL -- : Error caught: [ActionView::Template::Error] Couldn't find StorageResource without an ID
/Users/jeffreybonson/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5.1/lib/active_record/relation/finder_methods.rb:431:in `find_with_ids'
/Users/jeffreybonson/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5.1/lib/active_record/relation/finder_methods.rb:69:in `find'
/Users/jeffreybonson/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5.1/lib/active_record/associations/collection_association.rb:100:in `find'
/Users/jeffreybonson/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5.1/lib/active_record/associations/collection_proxy.rb:138:in `find'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/cloud_volume_helper/textual_summary.rb:59:in `textual_storage_resource'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:16:in `expand_textual_summary'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:47:in `block in expand_textual_group'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:47:in `map'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:47:in `expand_textual_group'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:53:in `expand_generic_group'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:73:in `block (2 levels) in process_textual_info'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:66:in `collect'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:66:in `block in process_textual_info'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:65:in `collect'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/helpers/textual_summary_helper.rb:65:in `process_textual_info'
/Users/jeffreybonson/Workspace/manageiq-ui-classic/app/views/layouts/_textual_groups_generic.html.haml:7:in `___sers_jeffreybonson__orkspace_manageiq_ui_classic_app_views_layouts__textual_groups_generic_html_haml__133123173389113515_745820'

```

jeffibm commented:
Also, can you please squash the commits.

Lauferin commented:
> Also, can you please squash the commits.

Sure.

jeffibm commented:
Hey @Lauferin , I am still finding the error mentioned in the above screenshot in my local env. I am not sure if I am missing anything or is there a dependent PR? It's looking for an id `StorageResource` which is not available for me. maybe providing a nil check would be better I believe!

Lauferin commented:
Hi @jeffibm , sorry for the delay.
I extended it to general exceptions that will also include if there is an id of a storage_resource that isn't available. If you still prefer me to only do a nil check let me know.

jeffibm commented:
Hey @Lauferin , it's good in local env. However, there are 3 recommendations from me.

1. In summary pages, if there is no data to be displayed, we can simply display a blank string instead of `Not Available`
2. A spec case if failing in `spec/shared/helpers/textual_summary_helper_methods.rb #29`. you need to add the `storage_resource` to the list from where it's being populated
3. Once this is done, we need to squash these commits to merge them.

Lauferin commented:
Hi @jeffibm
I returned nil, and it doesn't show that line at all which I think is good (it does the same when returning "").
I added the item to the list of the specs and squashed. I hope it's ok now.

Lauferin commented:
Hi @jeffibm 
Is there anything else to repair?
I see  that one of the checks is not successful but I can't figure out why.

jeffibm commented:
Hi @Lauferin , sorry for the late reply as i am on sick leave..
I believe the GH action can be fixed once you rebase with master..

Lauferin commented:
Hi @jeffibm
Rebased now :)
